core/unix/src/TUnixSystem.cxx

```diff
@@ -40,10 +40,15 @@
 #include <algorithm>
 #include <atomic>
 
+#ifdef __linux__
+#include <syscall.h>
+#endif
+
 //#define G__OLDEXPAND
 
 #include <unistd.h>
 #include <stdlib.h>
+#include <poll.h>
```

Comment:
`select` could work, but it may be better to just disable the timeouts on Windows; on RHEL7 with some common GCC options, select triggers an assert when the FD is >= 1024.

(Is there a signal-safe read function elsewhere in ROOT that has timeout?  Seems silly that it is only used here.)